src/sentry/services/hybrid_cloud/app/impl.py

```diff
@@ -253,3 +250,20 @@ def create_internal_integration_for_channel_request(
             installation = SentryAppInstallation.objects.get(sentry_app=sentry_app)
 
         return serialize_sentry_app_installation(installation=installation, app=sentry_app)
+
+    def prepare_sentry_app_components(
+        self,
+        *,
+        installation_id: int,
+        component_type: str,
+        project_slug: Optional[str] = None,
+        values_json: Optional[str] = None,
+    ) -> Optional[RpcSentryAppComponent]:
+        from sentry.models.integrations.sentry_app_installation import prepare_sentry_app_components
+
+        installation = SentryAppInstallation.objects.get(id=installation_id)
+        values = json.loads(values_json) if values_json else None
```

Comment:
Should we handle json parsing failures here? We are getting a JSON blob off the wire.

Reply:
That's a good idea, and one that we should probably generalize to other RPC methods. (Grep for `data_json` to see those cases.)

However, considering that it's not used by the one call site and may remain unused indefinitely, I'll just remove it. We can always add it later (with good validation) when we need to.